### PR TITLE
User use cases

### PR DESCRIPTION
**Pull request by PavelBrest**

### CODE REVIEW

File: LT.Core/Contracts/User/Commands/RegisterUser.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                                 .Length(4);
+
+            RuleFor(p => p.Name).NotEmpty();
+
+            RuleFor(p => p.EmailAddress).NotEmpty()
+                                        .EmailAddress();
+
+            RuleFor(p => p.Password).NotEmpty()
+                                    .MinimumLength(6)
+                                    .MaximumLength(25);
```
Comment (DzmitryBratchuk) [inline LT.Core/Contracts/User/Commands/RegisterUser.cs]: see in LT.Core/Contracts/User/Queries/LoginUser.cs

**Comment by DzmitryBratchuk** [comment]
It is necessary to add the ability to update user data (For example we need to change password)

File: LT.Core/Contracts/User/Queries/LoginUser.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    internal class LoginUserValidator : AbstractValidator<LoginUser>
+    {
+        public LoginUserValidator()
+        {
+            RuleFor(p => p.Login).NotEmpty()
+                                 .Length(4);
+
+            RuleFor(p => p.Password).NotEmpty()
+                                    .MinimumLength(6)
+                                    .MaximumLength(25);
```
[the remark this answers is not in the document]
Comment (PavelBrest) [reply LT.Core/Contracts/User/Queries/LoginUser.cs]: > Maybe use regex: RuleFor(p => p.Password).NotEmpty().Matches(@"^(?=.[a-z])(?=.[A-Z])(?=.*\d)[\S]{6,25}$");

I guess this regex looks better `^(?=.*[a-z])(?=.*[A-Z]{1,})(?=.*\d)[\S]{6,25}$`, isn't it?

[the remark this answers is not in the document]
Comment (DzmitryBratchuk) [reply LT.Core/Contracts/User/Queries/LoginUser.cs]: As you wish, they both work)

**Pull request merged**